src/uwtools/config/formats/base.py

```diff
@@ -22,15 +22,19 @@ class Config(ABC, UserDict):
     several configuration-file formats.
     """
 
-    def __init__(self, config_file: OptionalPath = None) -> None:
+    def __init__(self, config: Union[dict, OptionalPath] = None) -> None:
```

Comment:
Does it make sense for us to have a re-usable type for `Union[dict, OptionalPath]`?

Reply:
Good question. I thought about it too. But we have this one, and also `Union[dict, Config, OptionalPath]` and `Union[dict, YAMLConfig, OptionalPath]`, and `Union[dict, DefinitePath]`, and I didn't immediately come up with names that would be catch enough to not obscure the intent of these types. I thought it might be better to see the full type in the signature as a reminder about what the function/method is required to handle. I'd want that information to be obvious from any shorter type name. If anything comes to mind that you think captures that, I'm interested.